### PR TITLE
Added tests and updated requirements

### PR DESCRIPTION
**Pull request by andrewmoorman**
requirements.txt now includes update to scipy and leidenalg

new tests folder added with 7 unit tests for functions which rely on scipy/leidenalg

cluster function in cluster.py refactored to separate community detection functions for testing

Note: Maybe we should create a "version 1.5.7" branch as mentioned?

### CODE REVIEW

**Comment by hisplan** [comment]
I see two `RB_genes_human.txt`. Let's remove one of them. Maybe keep the one in the `data` folder?

**Comment by hisplan** [comment]
Hmm. I'm not sure if your 1.5.7 branch is fully synced with dpeerlab:1.5.7.

**Comment by andrewmoorman** [comment]
Should be okay now

**Comment by hisplan** [comment]
@armMSKCC a few comments:

Code
- Bump up the version to 1.5.7 in `version.py`
- Run Black on `test_cluster.py` (https://github.com/psf/black). You can also run this from VS Code or PyCharm.

Notebook
- Change Phenograph to PhenoGraph.
- Add `phenograph.__version__` to display PhenoGraph version along with Scanpy versions.
- `pip install scanpy==1.6.0` - the notebook might not run with the old versions.
- Rename `Phenograph_tutorial.ipynb` to `tutorial-pbmc3k.ipynb`
- Rename `example-01-using-random-data-points.ipynb` to `example-using-random-data-points.ipynb`

README.md
- Update release notes.
- Add a new section and describe how one can run pytest and resolve the issue we encountered.
- Mention and add a link to the new tutorial.

**Comment by hisplan** [comment]
@armMSKCC Forgot one thing. The notebook failed to load the RB file.

File: README.md [hunk 1]
```diff
@@ -49,6 +49,11 @@ If you use PhenoGraph in work you publish, please cite our publication:
 Release Notes
 -------------
 
+### Version 1.5.7
+
+* Updated leidenalg and scipy version requirements, revised parallel jaccard to support scipy==1.5.1, and created a test collection for use with pytest (see above)
```
Comment (hisplan) [inline README.md]: "see above" --> "see below" you mean? Maybe make a link to the unit test section?

File: README.md [hunk 2]
```diff
@@ -127,3 +151,11 @@ Troubleshooting
 
 * When using `PhenoGraph` inside a conda environment `leiden` takes longer to complete for larger samples compared to the system Python.
 
+### `numpy.ufunc` runtime warning when running pytest
+
+* When running unit tests, pytest may deliver the following warning `RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility`. This is caused by an incompatability between newer versions of the pytest module and an older version of Python. Check your Python version with:
+
+    python --version
```
Comment (hisplan) [inline README.md:159]: Put this into a preformatted box like this:
```
python --version
```

**Review by hisplan — CHANGES_REQUESTED** [review]
Minor comments

**Comment by hisplan** [comment]
I still don't see these in the Jupyter notebook:
- Add `phenograph.__version__` to display PhenoGraph version along with Scanpy versions.
- `!pip install scanpy==1.6.0` - the notebook might not run with the old versions.

**Pull request merged**